src/connectors/articleService.ts

```diff
@@ -334,14 +335,69 @@ export class ArticleService extends BaseService<Article> {
 
     const articleVersion = await this.models.create({
       table: 'article_version',
-      data: { ...data, ...newData } as Partial<ArticleVersion>,
+      data: { ...data, ...newData, description } as Partial<ArticleVersion>,
     })
+    this.latestArticleVersionLoader.clear(articleId)
     return articleVersion
   }
 
   public countArticleVersions = async (articleId: string) =>
     this.models.count({ table: 'article_version', where: { articleId } })
 
+  public findArticleVersions = async (
+    articleId: string,
+    { take, skip }: { take?: number; skip?: number } = {},
+    onlyContentChange = true
+  ) => {
+    const records = await this.knexRO('article_version')
+      .select('*', this.knexRO.raw('COUNT(1) OVER() ::int AS total_count'))
+      .from(
+        this.knexRO('article_version')
+          .where({ articleId })
+          .modify((builder) => {
+            if (onlyContentChange) {
+              builder.select(
+                '*',
+                this.knexRO.raw(
+                  'LAG(content_id, 1) OVER(order by id) AS last_content_id'
+                ),
+                this.knexRO.raw(
+                  'LAG(content_md_id, 1) OVER(order by id) AS last_content_md_id'
```

Comment:
the `order by id` is default `ASC` order, is it actually `first_content_md_id`?

the `ORDER BY id DESC` would be `last_content_md_id`

Reply:
I think here is right, as `LAG` returns value evaluated at the row that is offset rows **before** the current row within the partition ,  see https://www.postgresql.org/docs/8.4/functions-window.html

Reply:
maybe rename to `prev_content_id` for `LAG` and `next_content_id` for `LEAD`?